tests/test_pickup.py

```diff
@@ -1,148 +1,56 @@
-        if vcr.record_mode != "none":
-            time.sleep(0.1)
-        batch.refresh()
-
-    # Insure the shipments after purchase
-    if batch.state == "purchased":
-        for shipment in batch.shipments:
-            shipment.insure(amount=100)
-
-    pickup = easypost.Pickup.create(
-        address=pickup_address,
-        batch=batch,
-        reference="internal_id_1234",
-        min_datetime=noon_on_next_monday.isoformat(),
-        max_datetime=(noon_on_next_monday + ONE_DAY).isoformat(),
-        is_account_address=True,
-        instructions="Special pickup instructions",
-    )
-
-    assert pickup.pickup_rates != [], pickup.messages
-
-    pickup.buy(carrier=pickup.pickup_rates[0].carrier, service=pickup.pickup_rates[0].service)
+def test_pickup_create(one_call_buy_shipment, basic_pickup):
+    pickup = generate_pickup(one_call_buy_shipment, basic_pickup)
+
+    assert isinstance(pickup, easypost.Pickup)
+    assert str.startswith(pickup.id, "pickup_")
+    assert pickup.pickup_rates is not None
+
+
+@pytest.mark.vcr()
+def test_pickup_retrieve(one_call_buy_shipment, basic_pickup):
+    pickup = generate_pickup(one_call_buy_shipment, basic_pickup)
+
+    retrieved_pickup = easypost.Pickup.retrieve(pickup.id)
+
+    assert isinstance(retrieved_pickup, easypost.Pickup)
+    assert retrieved_pickup == pickup
 
 
 @pytest.mark.vcr()
-def test_single_pickup(noon_on_next_monday):
-    """Create a Shipment, buy it, and then buy a pickup for it"""
-
-    pickup_address = easypost.Address.create(
-        verify=["delivery"],
-        name="TAKASHI KOVACS",
-        company="EasyPost",
-        street1="2889 W ASHTON BLVD",
-        street2="SUITE 325",
-        city="Lehi",
-        state="UT",
-        zip="84042",
-        country="US",
-        phone="415-456-7890",
-    )
-
-    shipment = easypost.Shipment.create(
-        to_address={
-            "name": "Customer",
-            "street1": "8308 Fenway Rd",
-            "city": "Bethesda",
-            "state": "MD",
-            "zip": "20817",
-            "country": "US",
-        },
-        from_address=pickup_address,
-        parcel={"weight": 21.2},
-    )
-
-    shipment.buy(rate=shipment.lowest_rate("USPS", "Priority"), insurance=100.00)
-
-    pickup = easypost.Pickup.create(
-        address=pickup_address,
-        shipment=shipment,
-        reference="internal_id_1234",
-        min_datetime=noon_on_next_monday.isoformat(),
-        max_datetime=(noon_on_next_monday + ONE_DAY).isoformat(),
-        is_account_address=True,
-        instructions="Special pickup instructions",
-    )
-
-    assert pickup.pickup_rates != [], pickup.messages
-
-    pickup.buy(carrier=pickup.pickup_rates[0].carrier, service=pickup.pickup_rates[0].service)
+def test_pickup_buy(usps, one_call_buy_shipment, basic_pickup):
+    pickup = generate_pickup(one_call_buy_shipment, basic_pickup)
+
+    bought_pickup = pickup.buy(carrier=usps, service="Nextday")
```

Comment:
This should be `NextDay` (capital D) and could be a fixture just like the `usps_service` is (call it `pickup_service`)